src/Core/Services/AuthService.cs

```diff
@@ -486,7 +489,17 @@ private async Task<MasterKey> MakePreloginKeyAsync(string masterPassword, string
 
                     await _cryptoService.SetMasterKeyEncryptedUserKeyAsync(tokenResponse.Key);
 
-                    if (masterKey != null)
+                    var decryptOptions = await _stateService.GetAccountDecryptionOptions();
+                    if (decryptOptions?.TrustedDeviceOption != null)
+                    {
+                        var key = await _deviceTrustCryptoService.DecryptUserKeyWithDeviceKeyAsync(decryptOptions?.TrustedDeviceOption.EncryptedPrivateKey, decryptOptions?.TrustedDeviceOption.EncryptedUserKey);
+                        if (key != null)
+                        {
+                            await _cryptoService.SetUserKeyAsync(key);
+                        }
+                    }
+                    else if (masterKey != null &&
+                        (!string.IsNullOrEmpty(tokenResponse.KeyConnectorUrl) || !string.IsNullOrEmpty(decryptOptions?.KeyConnectorOption?.KeyConnectorUrl)))
```

Comment:
Can we add the `if (tokenResponse.KeyConnectorUrl != null)` above to this `else if` ladder? we can move the `SetMasterKeyEncryptedUserKeyAsync` above it.

We should also be checking for the `KeyConnectorUrl` on the `decyptOptions` as part of the key connector check right?